libraries/config/src/config.cpp

```diff
@@ -275,6 +275,19 @@ FullNodeConfig::FullNodeConfig(Json::Value const &string_or_object, Json::Value
 
   try {
     node_secret = dev::Secret(wallet["node_secret"].asString(), dev::Secret::ConstructFromStringType::FromHex);
+    if (!wallet["node_public"].isNull()) {
```

Comment:
Why do we even have this fields in wallet file if we really need only `secret`?

Reply:
I am not sure. It is not mandatory. It does not need to be there. But in a way I think it is useful for user to be able to get its address/public key this way. If we want community user to give us its public key/address, what would be better alternative?